api/graphql/mutations/participant.py

```diff
@@ -34,3 +66,76 @@ async def add_comment(
             entity=CommentEntityType.participant, entity_id=id, content=content
         )
         return GraphQLComment.from_internal(result)
+
+    @strawberry.mutation
+    async def update_participant(
+        self,
+        participant_id: int,
+        participant: ParticipantUpsertInput,
+        info: Info,
+    ) -> Annotated['GraphQLParticipant', strawberry.lazy('api.graphql.schema')]:
+        """Update Participant Data"""
+        from api.graphql.schema import GraphQLParticipant
+
+        connection = info.context['connection']
+        player = ParticipantLayer(connection)
+
+        participant.id = participant_id
+
+        upserted = await player.upsert_participant(
+            ParticipantUpsert.from_dict(strawberry.asdict(participant)).to_internal()
+        )
+        updated_participant = (await player.get_participants_by_ids([upserted.id]))[0]  # type: ignore [attr-defined]
+
+        return GraphQLParticipant.from_internal(updated_participant)
+
+    @strawberry.mutation
+    async def upsert_participants(
+        self,
+        participants: list[ParticipantUpsertInput],
+        info: Info,
+    ) -> list[Annotated['GraphQLParticipant', strawberry.lazy('api.graphql.schema')]]:
+        """
+        Upserts a list of participants with samples and sequences
+        Returns the list of internal sample IDs
+        """
+        from api.graphql.schema import GraphQLParticipant
+
+        connection: Connection = info.context['connection']
+        connection.check_access(FullWriteAccessRoles)
```

Comment:
Can't use `check_access` as it requires a project on the connection

Reply:
Fixed!